### PR TITLE
Add interoperability with ENV variables for uri and debug params #23

### PR DESCRIPTION
**Pull request by repli2dev**
Currently done for uri (most important) and debug (also quite important) params, others were tried but as the passthrough of the other parameters is done at compile time it was not done, because it would be incoherent with the evaluation of uri and debug params -- they are evaluated during runtime.

Due to this issue the panel is registered but is at runtime not included in tracy when debug mode is not enabled.

### CODE REVIEW

File: src/DI/RedisExtension.php [hunk 1]
```diff
@@ -150,9 +152,20 @@ public function afterCompile(ClassType $class): void
 	{
 		$config = $this->config;
 
+		// Add panel to tracy bar only if debug is allowed
 		if ($config->debug && $config->connection !== []) {
-			$initialize = $class->getMethod('initialize');
-			$initialize->addBody('$this->getService(?)->addPanel($this->getService(?));', ['tracy.bar', $this->prefix('panel')]);
+			$generator = new PhpGenerator($this->getContainerBuilder());
+			$this->initialization->addBody($generator->formatPhp('?;', [
```
Comment (f3l1x) [inline src/DI/RedisExtension.php:158]: Why is this needed?

Comment (repli2dev) [reply src/DI/RedisExtension.php:158]: This is needed in order to allow debug be set via ENV variable too... The given ENV node (Nette\DI\Statement) needs to be evaluated on the runtime properly (just passing it as another placeholder into the previous statement did not work as it did not unfolded from the Nette\DI\Statement.

Comment (f3l1x) [reply src/DI/RedisExtension.php:158]: I see. Thanks, I must take a look at it closer.

**Comment by repli2dev** [comment]
Any progress on this?

**Comment by f3l1x** [comment]
Hi @repli2dev, not yet. I will let you know. Thank you for understanding.

**Comment by repli2dev** [comment]
Kindly reminding this PL as this missing change blocks us in using contributte/redis in our Docker containers.

**Comment by f3l1x** [comment]
Hi @repli2dev. Can you please test current master?

URI should work pretty good with ENV. I would suggest use dynamic parameters instead of `::getenv`. It's more predictable way.

I am not sure about debug parameter. I think I need to discuss it with @dg, what is the best way to handle this type of use case. Because, every parameter can be dynamic at all. Not just uri or debug. Likewise, other packages in contributte should follow same concept. Should I do debug parameter dynamic for all? I think it's pretty clear we have to introduce some kind of logic. Maybe in nette/di or at least at contributte/di. :-)

**Comment by repli2dev** [comment]
Current master solves the issue enough. Closing for now.
